### DISCUSSION TITLE
[AsyncNotifier] maintain previous `data` state while new state is being built

### DISCUSSION THREAD

**Discussion by dJani97** [rrousselGit/riverpod#2034] [Q&A]
Each time an AsyncNotifier rebuilds, `AyncValue.loading()` is automatically emitted for the duration of the async `build()` function. This is great for most cases.

But what if, on some occasions, I don't want to emit a loading state while the notifier rebuilds, but instead, I'd prefer to keep the current state until the next one is loaded?

So, when the current state is `data`, when a watched provider changes, instead of emitting `loading`, I'd like to maintain the current `data` state until a new `data` or `error` state is emitted.

I was able to implement this in Cubits by doing something like this:

```dart

Future<void> load() async {
  state.maybeWhen(
    loaded: (_) {
      // do nothing
    },
    orElse: () {
      emit(const AyncValue.loading());
    },
  );
  
  // do the async operation
}
```

Is there any way to do this with riverpod?

**Comment by rrousselGit** [comment]
During the loading state, the previous state is still available under `AsyncValue.value`.

`when`/`map` and variants offer named parameters to skip the loading if you so wish.
Like `when(skipLoadingOnReload: true, ...)`
